merino-web/src/middleware/metrics.rs

```diff
@@ -23,13 +24,14 @@ pub struct MetricsMiddleware<S> {
     service: S,
 }
 
-impl<S> Transform<S, ServiceRequest> for Metrics
+impl<S, B> Transform<S, ServiceRequest> for Metrics
 where
-    S: Service<ServiceRequest, Response = ServiceResponse>,
+    S: Service<ServiceRequest, Response = ServiceResponse<B>>,
+    B: 'static + MessageBody,
     S::Future: 'static,
     S::Error: fmt::Debug,
 {
-    type Response = ServiceResponse;
+    type Response = S::Response;
```

Comment:
I had to make these changes to fix a compile error introduced by the crate upgrades, but I don't fully understand why this solution compiles (or if it's even correct)...I'd appreciate an extra set of 👀 

Same thing for the other `Transform` `impl`s I changed in this PR